azurerm/_modules/aks/node_pool/main.tf

```diff
@@ -0,0 +1,27 @@
+data "azurerm_kubernetes_cluster" "current" {
+  name                = var.cluster_name
+  resource_group_name = var.resource_group
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "current" {
+  name                  = var.node_pool_name
+  kubernetes_cluster_id = data.azurerm_kubernetes_cluster.current.id
+  enable_auto_scaling   = var.enable_auto_scaling
+  max_count             = var.enable_auto_scaling ? var.max_count : null
+  min_count             = var.enable_auto_scaling ? var.min_count : null
+  node_count            = var.node_count
+  vm_size               = var.vm_size
+  node_labels           = local.labels
+  node_taints           = local.taints
+  availability_zones    = var.availability_zones
+  max_pods              = var.max_pods
+  os_disk_type          = var.os_disk_type
+  priority              = var.use_spot ? "Spot" : "Regular"
+  eviction_policy       = var.use_spot ? var.eviction_policy : null
+  spot_max_price        = var.use_spot ? var.max_spot_price : null
+
+#   lifecycle {
+#       ignore_changes = [node_count]
+#   }
```

Comment:
I remember a while back having issues with `node_count` when autoscaling, as it's basically used for creation, but also used to be aligned when updating.

Ideally, azure provider should know that when autoscaling is on -> no need to pay attention to the actual node count.

As I never had fixed node pool, I used to have the `node_count` here, but right now I'm unsure if we want this.

Reply:
For EKS something similar was required, I also think it should be handled in the provider then but it isn't the case. The docs there say to use lifecycle ignore_changes. I will leave this note here for me to test this later.